contracts/escrow_fee_distributor/src/contract.rs

```diff
@@ -0,0 +1,478 @@
+            &VotingQueryMsg::TotalVotingPowerAt {
+                time: get_time_by_period(claim_period),
+            },
+        )?;
+
+        if user_voting_power.voting_power > Uint128::zero() {
+            claim_amount = claim_amount.checked_add(calculate_user_balance_per_week(
+                deps.as_ref(),
+                claim_period,
+                user_voting_power.voting_power,
+                total_voting_power.voting_power,
+            )?)?;
+        }
+
+        claim_period += 1;
+    }
+
+    CLAIM_FROM_PERIOD.save(deps.storage, receiver, &claim_period)?;
+
+    Ok(claim_amount)
+}
+
+/// ## Description
+/// Returns user amount for specified period
+fn calculate_user_balance_per_week(
+    deps: Deps,
+    period: u64,
+    user_vp: Uint128,
+    total_vp: Uint128,
+) -> StdResult<Uint128> {
+    let tokens_per_week = TOKENS_PER_WEEK
+        .may_load(deps.storage, U64Key::from(period))?
+        .unwrap_or_default();
+
+    Ok(user_vp.multiply_ratio(tokens_per_week, total_vp))
+}
+
+/// ## Description
+/// Updates general settings. Returns an [`ContractError`] on failure or the following [`Config`]
+/// data will be updated if successful.
+fn update_config(
+    deps: DepsMut,
+    info: MessageInfo,
+    max_limit_accounts_of_claim: Option<u64>,
+    is_claim_disabled: Option<bool>,
+) -> Result<Response, ContractError> {
+    let mut attributes = vec![attr("action", "update_config")];
+    let mut config: Config = CONFIG.load(deps.storage)?;
+
+    if info.sender != config.owner {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    if let Some(is_claim_disabled) = is_claim_disabled {
+        config.is_claim_disabled = is_claim_disabled;
+        attributes.push(Attribute::new(
+            "is_claim_disabled",
+            is_claim_disabled.to_string(),
+        ));
+    };
+
+    if let Some(max_limit_accounts_of_claim) = max_limit_accounts_of_claim {
+        config.max_limit_accounts_of_claim = max_limit_accounts_of_claim;
+        attributes.push(Attribute::new(
+            "max_limit_accounts_of_claim",
+            max_limit_accounts_of_claim.to_string(),
+        ));
+    };
+
+    CONFIG.save(deps.storage, &config)?;
+
+    Ok(Response::new().add_attributes(attributes))
+}
+
+/// ## Description
+/// Available the query messages of the contract.
+/// ## Queries
+/// * **QueryMsg::UserFeeAmountPerWeek { user, timestamp }** Returns the user fee amount by
+/// the timestamp.
+///
+/// * **QueryMsg::Config {}** Returns the base controls configs that contains in the [`Config`]
+/// object.
+///
+/// * **QueryMsg::FeeTokensPerWeek { start_after, limit }** Returns the vector with the amount of
+/// tokens for the week distribution with specified parameters.
+///
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    match msg {
+        QueryMsg::UserFeeAmountPerWeek { user, timestamp } => {
```

Comment:
Please be consistent in namings, the query enum is called UserFeeAmountPerWeek, the method is called query_user_balance and the variable it returns is called user_fee_balance. So what does this function do? It's hard to quickly understand because of this inconsistency

Reply:
Also please check other query functions, suck as FeeTokensPerWeek